server/svix-server/src/core/types.rs

```diff
@@ -603,6 +616,35 @@ impl Validate for EndpointHeaders {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Default)]
+pub struct EndpointHeadersPatch(pub HashMap<String, Option<String>>);
+json_wrapper!(EndpointHeadersPatch);
+
+impl<'de> Deserialize<'de> for EndpointHeadersPatch {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        HashMap::deserialize(deserializer)
+            .map(|x: HashMap<String, Option<String>>| x.into_iter().map(|(k, v)| (k, v)).collect())
```

Comment:
Does the second map actually do anything?

Reply:
Nope! Although I think this was here before I came along. Patched.